x/oracle/abci.go

```diff
@@ -70,26 +71,50 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) error {
 
 		//---------------------------
 		// Do miss counting & slashing
-		voteTargetsLen := len(voteTargets)
+		denomMap := map[string]map[string]struct{}{}
+
+		for _, denom := range voteTargets {
+			denomMap[denom] = map[string]struct{}{}
+		}
 
+		for denom, votes := range voteMap {
+			for _, vote := range votes {
+				// ignore denoms, not requested in voteTargets
+				_, ok := denomMap[denom]
+				if !ok {
+					continue
+				}
+
+				denomMap[denom][vote.Voter.String()] = struct{}{}
+			}
+		}
+
+		// Check if each validator is missing a required denom price
 		for _, claim := range validatorClaimMap {
-			// Skip abstain & valid voters
-			if int(claim.WinCount) >= voteTargetsLen {
-				continue
+
+			missing := false
+			for _, denom := range voteTargets {
+				_, ok := denomMap[denom][claim.Recipient.String()]
+				if !ok {
+					missing = true
+					break
+				}
 			}
 
-			// Increase miss counter
-			k.SetMissCounter(ctx, claim.Recipient, k.GetMissCounter(ctx, claim.Recipient)+1)
+			if missing {
+				// Increase miss counter
+				k.SetMissCounter(ctx, claim.Recipient, k.GetMissCounter(ctx, claim.Recipient)+1)
+			}
 		}
 
-		// Distribute rewards to ballot winners
-		k.RewardBallotWinners(
-			ctx,
-			(int64)(params.VotePeriod),
-			(int64)(params.RewardDistributionWindow),
-			voteTargets,
-			validatorClaimMap,
-		)
+		// // Distribute rewards to ballot winners
+		// k.RewardBallotWinners(
```

Comment:
@starsquidnodes except for the fact that we don't have a reward pool here, is there a reason to remove this? might be nice to leave the option in, in case we wanted to fund the pool to pay for API subscriptions if ever needed for example

Reply:
Yes, the idea was to stop further calculations because we don't use rewards